### ISSUE TITLE
Bug using hash and array helpers together

### ISSUE THREAD

**Issue by mansona** [emberjs/ember.js#18693]
There seems to be a bug that was introduced in ember-source@3.17.0-beta.1 🤔but interestingly it doesn't seem to have been happening on ember-canary before the beta was released

Essentially it looks like using the `{{array}}` and `{{hash}}` helpers together doesn't work with the error: 

```
TypeError: Cannot convert object to primitive value
```

I have produced a replication and run tests with ember try here: https://github.com/mansona/bug-hash-array-reproduction/pull/1 and the build is here https://travis-ci.com/mansona/bug-hash-array-reproduction/builds/145761580

You will see that it fails on ember-beta and ember-canary.

Here is the example of the template that fails: 

```
<table>
  <tbody>
    {{#each
      (array
        (hash value='' name='all')
        (hash value='x' name="horizontal")
        (hash value='y' name="vertical")
        (hash value='t' name="top")
        (hash value='r' name="right")
        (hash value='b' name="bottom")
        (hash value='l' name="left")
      ) as |item|
    }}

      <tr>
        <td>{{item.value}}</td>
        <td>{{item.name}}</td>
      </tr>
    {{/each}}
  </tbody>
</table>
```
The table is not required, I just put it in there to make it clear what an example app might be 👍

**Comment by rwjblue** [comment]
Hmm, thank you for reporting! Definitely seems like a regression to me.

**Comment by mansona** [comment]
Has there been any update on this? I see that we've cut a new beta but it doesn't seem to have changed anything 🤔

**Comment by pzuraq** [comment]
Tracked down the root cause and have opened up a PR to fix upstream in the VM: https://github.com/glimmerjs/glimmer-vm/pull/1022

**Comment by mansona** [comment]
thanks a million for fixing this @pzuraq 🎉 I'm assuming this will be backported into the 3.17 release branch then once it's fixed so that beta will start working again?

**Comment by mansona** [comment]
It looks like this has started working now 🎉 I'm assuming the fix was included in v3.17.0-beta.5 then?

Thanks everyone for working on this

**Comment by chancancode** [comment]
@mansona I don't think so, the only change in beta 5 is docs
